src/instance.rs

```diff
@@ -1876,10 +1875,21 @@ where
 impl<T> std::fmt::Display for Py<T>
 where
     T: PyTypeInfo,
-    T::AsRefTarget: std::fmt::Display,
 {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        Python::with_gil(|py| std::fmt::Display::fmt(self.bind(py), f))
+        Python::with_gil(|py| {
+            use crate::types::PyAnyMethods;
+            let obj = self.bind(py).as_any();
+            match obj.str() {
+                Ok(s) => return f.write_str(&s.to_string_lossy()),
+                Err(err) => err.write_unraisable_bound(py, Some(obj)),
+            }
+
+            match obj.get_type().name() {
+                Ok(name) => write!(f, "<unprintable {} object>", name),
+                Err(_err) => f.write_str("<unprintable object>"),
+            }
+        })
```

Comment:
Did this implementation need expanding? It looks like it already went through `Bound`...

Reply:
Ah, good catch, we can just drop the `AsRefTarget` bound 👍